scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -171,6 +171,24 @@
                       (catch Exception e
                         (log/debug e "Error reading a string from mesos status data. Is it in the format we expect?")))))})
 
+(defn update-reason-histos!
+  "Updates histograms for run time, cpu time, and memory time,
+  where the histograms have the failure reason in the title"
+  [db mesos-reason instance-runtime {:keys [cpus mem]}]
+  (let [reason (->> mesos-reason
+                    (reason/mesos-reason->cook-reason-entity-id db)
+                    (d/entity db)
+                    :reason/name
+                    name)
+        update-histo! (fn update-histo! [s v]
+                        (histograms/update!
+                          (histograms/histogram
+                            ["cook-mesos" "scheduler" (str "hist-task-fail-" s "-reason-" reason)])
```

Comment:
Can we create nesting on reason and then the times?

Reply:
Sure thing. Here's what they look like now:

```bash
cook-mesos.scheduler.hist-task-fail.mesos-command-executor-failed.cpu-times
             count = 1
               min = 402
               max = 402
              mean = 402.00
            stddev = 0.00
            median = 402.00
              75% <= 402.00
              95% <= 402.00
              98% <= 402.00
              99% <= 402.00
            99.9% <= 402.00
cook-mesos.scheduler.hist-task-fail.mesos-command-executor-failed.mem-times
             count = 1
               min = 51456
               max = 51456
              mean = 51456.00
            stddev = 0.00
            median = 51456.00
              75% <= 51456.00
              95% <= 51456.00
              98% <= 51456.00
              99% <= 51456.00
            99.9% <= 51456.00
cook-mesos.scheduler.hist-task-fail.mesos-command-executor-failed.times
             count = 1
               min = 402
               max = 402
              mean = 402.00
            stddev = 0.00
            median = 402.00
              75% <= 402.00
              95% <= 402.00
              98% <= 402.00
              99% <= 402.00
            99.9% <= 402.00
```